collector/plugins/sources/prometheus/prometheus.go

```diff
@@ -189,13 +190,14 @@ func (src *prometheusMetricsSource) parseMetrics(reader io.Reader) ([]wf.Metric,
 	for !metricReader.Done() {
 		var parser expfmt.TextParser
 		reader := bytes.NewReader(metricReader.Read())
+
+		// TODO bug: err is overwritten here for every metric,
+		// so whatever happens to be the last value of err is what is returned
```

Comment:
Should we fix the error message to accommodate multiple failures? Perhaps using something like [`errors.Join`](https://pkg.go.dev/errors#Join)?